instruments/minghe/mhs5200a.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides the support for the MingHe low-cost function generator.
+
+Class originally contributed by Catherine Holloway.
+"""
+
+# IMPORTS #####################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+from builtins import range
+from enum import Enum
+
+import quantities as pq
+
+from instruments.abstract_instruments import Instrument
+from instruments.util_fns import ProxyList, assume_units
+
+# CLASSES #####################################################################
+
+
+class MHS5200(Instrument):
+    """
+    The MHS5200 is a low-cost, 2 channel function generator.
+
+    There is no user manual, but Al Williams has reverse-engineered the
+    communications protocol:
+    https://github.com/wd5gnr/mhs5200a/blob/master/MHS5200AProtocol.pdf
+    """
+    # pylint: disable=unused-variable
+    def __init__(self, filelike):
+        super(MHS5200, self).__init__(filelike)
+        self._channel_count = 2
+
+    # INNER CLASSES #
+
+    class Channel(object):
```

Comment:
`_get_amplitude_` and `_set_amplitude_` are there because we have a concrete implementation for `amplitude` in `FunctionGenerator`. This implementation is what gets called first, and handles conversion for volts vpp vs rms, and dBm. For setting, the numerical value, and the units, are then passed to `_set_amplitude_` so that you can then set things correctly on the instrument.

This lets you do this:

```python
>>> inst.amplitude = (5 * pq.V, inst.VoltageMode.rms)
>>> inst.amplitude = (1 * pq.V, inst.VoltageMode.peak_to_peak)
>>> inst.amplitude = 7 * pq.V  # assumes vpp
>>> inst.amplitude = 1 * ik.units.dBm
```